src/lib/features/feature-lifecycle/feature-lifecycle-service.ts

```diff
@@ -18,47 +20,71 @@ export class FeatureLifecycleService {
 
     private environmentStore: IEnvironmentStore;
 
-    constructor({
-        eventStore,
-        featureLifecycleStore,
-        environmentStore,
-    }: {
-        eventStore: IEventStore;
-        environmentStore: IEnvironmentStore;
-        featureLifecycleStore: IFeatureLifecycleStore;
-    }) {
+    private flagResolver: IFlagResolver;
+
+    constructor(
+        {
+            eventStore,
+            featureLifecycleStore,
+            environmentStore,
+        }: {
+            eventStore: IEventStore;
+            environmentStore: IEnvironmentStore;
+            featureLifecycleStore: IFeatureLifecycleStore;
+        },
+        { flagResolver }: Pick<IUnleashConfig, 'flagResolver'>,
+    ) {
         this.eventStore = eventStore;
         this.featureLifecycleStore = featureLifecycleStore;
         this.environmentStore = environmentStore;
+        this.flagResolver = flagResolver;
+    }
+
+    private async checkEnabled(fn: () => Promise<void>) {
```

Comment:
each event needs to be wrapped with a check because we want runtime control. If we put this around the listen method we'd only have startup control.